corehq/blobs/models.py

```diff
@@ -4,3 +4,17 @@
 class BlobMigrationState(models.Model):
     slug = models.CharField(max_length=20, unique=True)
     timestamp = models.DateTimeField(auto_now=True)
+
+
+class BlobExpiration(models.Model):
+    '''
+    This models records when temporary blobs should be deleted
+    '''
+    bucket = models.CharField(max_length=255)
+    identifier = models.CharField(max_length=255)
+    expires_on = models.DateTimeField(db_index=True)
+    created_on = models.DateTimeField(auto_now=True)
+    length = models.IntegerField()
+
+    # This is set to True when the blob associated has been successfully deleted
+    deleted = models.BooleanField(default=False, db_index=True)
```

Comment:
Do these get cleaned up? what's the benefit of keeping these around after they're deleted?

Reply:
I don't think this discussion answers the question of why we keep these records around rather than just deleting them. I don't see a benefit to having them.

Reply:
> the fsdb blocks blobs from being updated if the path already exists

this [just changed today](https://github.com/dimagi/commcare-hq/pull/19936), btw. s3db never prevented that because it couldn't be done efficiently.

> I don't think this discussion answers the question of why we keep these records around rather than just deleting them. I don't see a benefit to having them.

I agree it doesn't really answer that question, but I can see some benefit from a historical record/logging perspective. Answering the question: why was this blob deleted? However, it comes up pretty rarely and I've never actually looked here to answer that question (silly mistake on my part, nothing more). I think ideally I'd like to delete them after 30 days or something, but that requires more machinery and code to maintain, so maybe not worth the extra effort over just deleting the record immediately? Maybe we should simply log that the blob was deleted (along with it's `BlobExpiration` object) and let log rotation take care of the rest?

Reply:
@snopoke I've added a task to my todo list. I'll delete them and add some logging.